addons/website_slides/static/src/js/slides_upload.js

```diff
@@ -373,13 +374,10 @@ var SlideUploadDialog = Dialog.extend({
      */
     // TODO: Remove this part, as now SVG support in image resize tools is included
     //Python PIL does not support SVG, so converting SVG to PNG
-    _svgToPng: function () {
+    _svgToPng: async function () {
```

Comment:
Ok, obviously this is annoying: you cannot change the signature of function in stable versions (unless *maybe* to add new optional parameters at the end and even that is to avoid). You always have to think about custo made by users / community modules and/or our own custo.

In fact, going deeper checking this code, I wonder if this is necessary at all :/ When I said https://github.com/odoo/odoo/pull/74956#discussion_r693973323 to you, it was a real question: is it necessary? :thinking: Here checking the call stack of that _svgToPng from website_slides, I see this seems only used for svg coming from previews of youtube videos and google documents -> they probably are not concerned by the bug on Firefox or we would have seen it I guess. Please double-check but so does not seem needed :grimacing:

Reply:
If you upload a problematic svg in a course it won't work. This is probably not a very common use case but it still a use case... In all the cases if we can't change the function signature we should keep this fix for master. Except if you see a way to avoid changing signature but I don't :s
![Screenshot from 2021-08-31 09-17-12](https://user-images.githubusercontent.com/49784121/131459815-9d0ad814-0198-48a3-b35d-d0d3610b82f2.png)

Reply:
@smottet 

> If you upload a problematic svg in a course it won't work.

Indeed, you are right, seems like I lost myself in the code yesterday. 

> Except if you see a way to avoid changing signature but I don't :s

I push-forced the branch with the solution. Could you have a look? Tell me if you see something wrong and maybe test on your side :+1:

Reply:
@smottet Forgot to tell you: did not find quickly how to test the part of the code where:
- `values['slide_type'] === 'webpage'` -> I can choose a SVG indeed but cannot see where it is used afterwards
- `/^image\/.*/.test(this.file.type)` + NOT `values['slide_type'] === 'presentation'` -> Cannot even see when we go that road

Can you check?

Reply:
@qsm-odoo  nice work around :clap: ! Works fine for me. It makes the code quite more complex but no choice if we want to fix it in stable...

> /^image\/.*/.test(this.file.type) + NOT values['slide_type'] === 'presentation' -> Cannot even see when we go that road

You must install survey. Then you can add slide of type 'certification' and upload images. But in the two cases you cited I can't find where the uploaded images are use afterwards (except on the slide.slide form views). On the client side I don't find any place where they are used...

Reply:
It's used as image when sharing to social media :wink: Let's merge